src/fields/lib/abstract-field.js

```diff
@@ -236,4 +247,80 @@ module.exports = class AbstractField {
       value === 'undefined' ? undefined :
         value;
   }
+
+  /**
+   * Checks for read permission on the field. Override or pass in constructor options
+   * this method for field level authentication. Defaults to true otherwise
+   *
+   * @method canRead
+   * @param bundle
+   * @returns {Promise.<boolean>}
+   */
+  canRead(bundle) {
+    if (this._canRead) {
+      return Promise.resolve(this._canRead.call(null, bundle));
+    }
+    return Promise.resolve(true);
+  }
+
+  /**
+   * Checks for create permission on the field. Override or pass in constructor options
+   * this method for field level authentication. Defaults to true otherwise
+   *
+   * @method canWriteOnCreate
+   * @param bundle
+   * @returns {Promise.<boolean>}
+   */
+  canWriteOnCreate(bundle) {
+    if (this._canWriteOnCreate) {
+      return Promise.resolve(this._canWriteOnCreate.call(null, bundle));
+    }
+    return Promise.resolve(true);
+  }
+
+  /**
+   * Checks for update permission on the field. Override or pass in constructor options
+   * this method for field level authentication. Defaults to true otherwise
+   *
+   * @method canWriteOnUpdate
+   * @param bundle
+   * @returns {Promise.<boolean>}
+   */
+  canWriteOnUpdate(bundle) {
+    if (this._canWriteOnUpdate) {
+      return Promise.resolve(this._canWriteOnUpdate.call(null, bundle));
+    }
+    return Promise.resolve(true);
+  }
+
+  /**
+   * Checks for all permissions requested for the field
+   *
+   * @param requestedPermissions
+   * @param bundle
+   * @returns {Promise.<boolean>}
+   */
+  authenticatePermissions(requestedPermissions, bundle) {
+    return Promise.all(requestedPermissions.map(perm => {
+      let permissionPromise;
+      switch (perm) {
+        case PermissionTypes.READ:
+          permissionPromise = this.canRead(bundle);
+          break;
+        case PermissionTypes.CREATE:
+          permissionPromise = this.canWriteOnCreate(bundle);
+          break;
+        case PermissionTypes.UPDATE:
+          permissionPromise = this.canWriteOnUpdate(bundle);
+          break;
+        default:
+          permissionPromise = Promise.resolve(true);
```

Comment:
I think I would be better if we don't block access (by throwing) for permissions we don't support